src/main/java/org/kairosdb/core/http/rest/QueryAutocompleter.java

```diff
@@ -0,0 +1,49 @@
+package org.kairosdb.core.http.rest;
+
+import com.google.common.collect.SetMultimap;
+import org.kairosdb.core.datastore.QueryMetric;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class QueryAutocompleter {
+    private static final Logger logger = LoggerFactory.getLogger(QueryAutocompleter.class);
+
+    public void complete(QueryMetric query) {
+        completeMetricTag(query);
+    }
+
+    private void completeMetricTag(QueryMetric query) {
+        final SetMultimap<String, String> tags = query.getTags();
+        final Set<String> keys = tags.get("key");
+        final Set<String> metrics = new HashSet<>();
+        for (final String key : keys) {
+            try {
```

Comment:
As I understood, the exception could be raised only inside the `extractMetricName` method. What do you think to put the `try/catch` block inside that method?

Reply:
We have to stop processing of the whole `completeMetricTag` method, and that's why we need exception handling on this level

Reply:
It totally make sense, but when the `null` is returned, you are also quitting the method. So it will be just one `return;` instead of two. But, as I said, it's not a big deal.

Reply:
I see your point. Agree, probably in terms of readability it would be better to move try/catch inside the method and return null in case of exception. NB: please make sure that exception is logged